core/templates/base-components/i18n-language-selector.component.ts

```diff
@@ -47,6 +47,9 @@ export class I18nLanguageSelectorComponent {
     this.currentLanguageCode = this.i18nLanguageCodeService
       .getCurrentI18nLanguageCode();
     this.supportedSiteLanguages = AppConstants.SUPPORTED_SITE_LANGUAGES;
+    this.i18nLanguageCodeService.onI18nLanguageCodeChange.subscribe((code) => {
+      this.currentLanguageCode = code;
+    });
```

Comment:
I have set up a two-way data flow between the selector component and i18nLanguageCodeService.

Earlier, when changing the language in the selector, we could update i18nLanguageCodeService but not the other way around. Sometimes, there may be a situation where the language of the site is changed by code (by some service, component), not using this selector.
For such cases, earlier, the selector was not getting updated.

The crux is that I have set up i18n in the codebase such that i18nLanguageService acts as a single source of truth for the language code. Changing the value stored in this service will auto-trigger changes in angular-translate, ngx-translate and any other component / service / directive linked to / using language code.